### DISCUSSION TITLE
colors in table input sql statement

### DISCUSSION THREAD

**Discussion by campen** [apache/hop#4499] [General]
Hey all!  I am checking out the latest 2.11 build and the colors in the sql new look great, but it takes quite a bit to display (spinning beach ball)...is this a known issue?

**Comment by bamaer** [comment]
Hi Dave, 

I just tested in a local build on MacOS. Even though there may be a slight delay in opening Table Input transforms, I haven't seen any transforms that take longer than one second to open.
Can you provide some more details on your environment: OS, java version, Hop build, RDBMS type and cloud vs on-prem etc. Also, can you provide a comparison of the time it takes to open an average Table Input with vs without syntax highlighting?

**Comment by xdelox** [comment]
Hi Dave, let me jump in, as I'm interested in this topic, somehow.

I don't think it's a matter of OS or Java version. Instead, the delay
you're experiencing might be due to reserved words and function/procedure
names being retrieved from a query against the connection's database.

This means different users may experience a wide range of performance,
depending on various factors like network traffic, bandwidth, server load,
and so on.

Can you please just measure the difference with a DB on your localhost,
assuming that your target DB is not on your machine?

Il giorno mer 30 ott 2024 alle ore 07:54 Bart Maertens <
***@***.***> ha scritto:

**Comment by bamaer** [comment]
agreed, if we want to extend the syntax highlighting to include even more database metadata (`information_schema`) etc, we could consider including an `enable_syntax_highlighting` flag or something similar on the database level.

**Comment by campen** [comment]
I see the issue now...this is not related to the colors. it looks like the table input looks for a valid db connection and if that is not valid, it spins and takes about 20 seconds to open. If I have a valid connection, it doesn't take any time at all.  This did not happen until 2.11.

I was only connecting to a local (on my machine) Docker containerized Postgres db.

**Reply by hansva** [reply]
Thanks for the information.
I would suggest we take the following actions and see if it improves the situation:
- Do not try to load the keywords when the connection field is empty (seems this is happening)
- Do not try to load the keywords when the connection is a variable (might not have an actual value causing a search for non-existing keywords)

**Reply by hansva** [reply]
@campen Are you able to build Hop from a branch or do you need a published zip to test something?

**Reply by campen** [reply]
I can build Hop from a branch - I just need to know the branch to pull

**Reply by hansva** [reply]
Great, I have made https://github.com/apache/hop/pull/4505
This is a draft and I only changed Table Input to see if this resolves your issues. If it does I will push it to all locations.

**Comment by campen** [comment]
After realizing this is due to an invalid db connection, and the behavior was introduced in v 2.11 with the syntax highlighting. I wrote up a new ticket to give a popup dialog with the reason why the db connection failed.  https://github.com/apache/hop/issues/4506
